File: csharp/Original/PublishingHouse.Persistence/Books/BookEntity.cs
using PublishingHouse.Persistence.Authors;
using PublishingHouse.Persistence.Books.Entities;
using PublishingHouse.Persistence.Books.ValueObjects;
using PublishingHouse.Persistence.Publishers;
using PublishingHouse.Persistence.Reviewers;

namespace PublishingHouse.Persistence.Books;

public class BookEntity
{
    public enum State { Writing, Editing, Printing, Published, OutOfPrint }

    public required Guid Id { get; set; }
    public required State CurrentState { get; set; }
    public required string Title { get; set; }
    public required AuthorEntity Author { get; set; }
    public string? Genre { get; set; }
    public required PublisherEntity Publisher { get; set; }
    public int Edition { get; set; }
    public string? ISBN { get; set; }
    public DateOnly? PublicationDate { get; set; }
    public int? TotalPages { get; set; }
    public int? NumberOfIllustrations { get; set; }
    public string? BindingType { get; set; }
    public string? Summary { get; set; }
    public CommitteeApproval? CommitteeApproval { get; set; }
    public required List<ReviewerEntity> Reviewers { get; set; } = new();
    public required List<ChapterEntity> Chapters { get; set; } = new();
    public required List<Translation> Translations { get; set; } = new();
    public required List<Format> Formats { get; set; } = new();
}

File: node.js/src/slimmed/persistence/books/bookEntity.ts
import { AuthorEntity } from '../authors';
import { PublisherEntity } from '../publishers/publisherEntity';
import { ReviewerEntity } from '../reviewers';
import { ChapterEntity, FormatEntity } from './entities';
import { TranslationEntity } from './entities/translationEntity';
import { CommitteeApprovalVO } from './valueObjects';

export type BookEntity = {
  id: string;
  currentState: State;
  title: string;
  author: AuthorEntity;
  publisher: PublisherEntity;
  edition: number;
  genre?: string | null;
  isbn?: string | null;
  publicationDate?: Date | null;
  totalPages?: number | null;
  numberOfIllustrations?: number | null;
  bindingType?: string | null;
  summary?: string | null;
  committeeApproval?: CommitteeApprovalVO | null;
  reviewers: ReviewerEntity[];
  chapters: ChapterEntity[];
  translations: TranslationEntity[];
  formats: FormatEntity[];
  version: number;
};

export enum State {
  Writing = 'Writing',
  Editing = 'Editing',
  Printing = 'Printing',
  Published = 'Published',
  OutOfPrint = 'OutOfPrint',
}
